bittensor/_neuron/text/core_validator/__init__.py

```diff
@@ -543,6 +563,7 @@ def neuron_stats_update(self, neuron_stats: Dict[int, Dict[str, Any]]):
                 zkey = key + '!'  # zeroing key
                 stats.setdefault(zkey, 0.)  # initialize zkey val to zero to gradually increase with observations
                 if key in _stats and not math.isnan(_stats[key]):
+                    responsive_uids += [_uid]
```

Comment:
Does moving this line up change anything?

Reply:
Yes, it will now include a responsive UID also on first encounter, where previously it required that a UID already existed in `neuron_stats`. So this fixes it.